DigitalLearningSolutions.Web.Tests/Controllers/Login/LoginControllerTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.Login
+{
+    using DigitalLearningSolutions.Data.Exceptions;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.Helpers;
+    using DigitalLearningSolutions.Web.Controllers;
+    using DigitalLearningSolutions.Web.Tests.TestHelpers;
+    using DigitalLearningSolutions.Web.ViewModels.Login;
+    using FakeItEasy;
+    using FluentAssertions.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    internal class LoginControllerTests
+    {
+        private LoginController controller;
+        private ILoginService loginService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            loginService = A.Fake<ILoginService>();
+            controller = LoginTestHelper.GetLoginControllerWithUnauthenticatedUser(loginService);
+        }
+
+        [Test]
+        public void Index_should_render_if_user_is_unauthenticated()
+        {
+            // When
+            var result = controller.Index();
+
+            // Then
+            result.Should().BeViewResult().WithDefaultViewName();
+        }
+
+        [Test]
+        public void Index_should_redirect_if_user_is_authenticated()
+        {
+            // Given
+            var controllerWithAuthenticatedUser = LoginTestHelper.GetLoginControllerWithAuthenticatedUser(loginService);
+
+            // When
+            var result = controllerWithAuthenticatedUser.Index();
+
+            // Then
+            result.Should().BeRedirectToActionResult().WithControllerName("Home").WithActionName("Index");
+        }
+
+        [Test]
+        public void Successful_sign_in_should_render_home_page()
+        {
+            // Given
+            A.CallTo(() => loginService.VerifyUserDetailsAndGetClaims(A<string>._, A<string>._)).Returns((UserTestHelper.GetDefaultAdminUser(), UserTestHelper.GetDefaultDelegateUser()));
```

Comment:
Really minor thing across the MR - some of the lines do get quite long with the mocks, so it might make it slightly easier to read if the "Returns" is put on a new line :)

Reply:
I thought I had done that but it turns out that Resharper's Cleanup Code puts them back by default. I'll manually change them back.